modules/programs/ripgrep.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.ripgrep;
+in {
+  meta.maintainers = [ hm.maintainers.pedorich-n ];
+
+  options = {
+    programs.ripgrep = {
+      enable = mkEnableOption "Ripgrep";
+
+      package = mkPackageOption pkgs "ripgrep" { };
+
+      configDir = mkOption {
+        type = types.str;
+        default = "${config.xdg.configHome}/ripgrep";
+        defaultText =
+          literalExpression ''"''${config.xdg.configHome}/ripgrep"'';
+        description = ''
+          Directory where the <filename>ripgreprc</filename> file will be stored.
+        '';
+      };
```

Comment:
I would suggest avoiding this type of option in the future and IMHO I would prefer removing this option before too many use the module.

Basically, we should use the default configuration location as much as possible. We have tried to force programs to use XDG configuration directories before and it tends to cause issues. Especially when one has to involve environment variables, which causes the program not to find the configuration unless the user relogs.

Instead please convince upstream to support the XDG configuration directory.

Reply:
Discussed in https://github.com/nix-community/home-manager/pull/4017#discussion_r1206968471

Reply:
Aha, ok. I'll move there.